enhancements/multi-arch/multiarch-manager-operator.md

```diff
@@ -0,0 +1,799 @@
+When pulling the images, the access to registries is done by merging (1) the
+global pull secret stored within a node, and (2) all the `dockerconfigjson`
+secrets listed by the pod spec's `ImagePullSecrets` list. The `ImagePullSecrets`
+fields are either provided by the users or filled automatically by the
+Kubernetes Pod Admission controllers through the ones associated with the
+service account (SA) running the pod.
+
+##### Registry reachability (proxy and mirroring)
+In phase 2, the `RegistriesMirrorCacher` will watch ICSPs/IDMSs to sync the
+registry mirror configurations within a map having a `source` as key and the
+list of `mirrors` as values.  The inspection of the images will change such that
+if the map has keys available, before trying to reach the original URI, the URI
+will be looked up in the map (the registry base URL only). If a hit occurs, the
+mirrors will be tried in order to inspect the image until the inspection
+succeeds or the list is fully visited. Finally, no changes are expected to be
+applied to support configurations like cluster-wide proxy at the time of
+writing.
+
+##### Registry and image verification
+
+The `ImageRegistryController` watches the `image.config.openshift.io/cluster`
+CR[^7]. It caches and adapts the image registries'
+configuration stored in the `image.config.openshift.io/cluster` object, to set
+the `SystemConfig` of the image inspector with the information about 
+verifying the registries.
+
+The CA bundles used to verify the TLS connections to the registries are:
+
+- Global CA bundle: a ConfigMap local to the namespace where the operator is deployed is created and labeled
+  with `config.openshift.io/inject-trusted-cabundle=true`, so that it stays in sync with the
+  proxy/cluster's `spec.trustedCA` config map reference and the global CA bundle deployed in the nodes. It is mounted in
+  the controller pod at `/etc/pki/tls/certs/ca-bundle.crt` so that the crypto Golang package will source it[^9] to
+  perform the TLS handshake and verification. The operator watches the ConfigMap and redeploys the Pod placement
+  controller if it changes.
+- Additional CA bundles: these are inferred from the configmap `image-registry-certificates` in
+  the `openshift-image-registry` namespace, to track any additional CA bundles set in
+  the `image.config.openshift.io/cluster` object and only used to verify image registries defined within the ConfigMap.
+  The pod placement controller watches this config map for updates and patches the `/etc/docker/certs.d` directory
+  according to changes in the ConfigMap, similar to the node-ca DaemonSet managed by the image registry operator[^9].
+- Finally, the Pod Placement Controller watches the `image.config.openshift.io/cluster`[^7] and patches
+  the`/etc/containers/policy.json` and `/etc/containers/registries.conf` files in the pod placement controller according
+  to the settings of the `image.config.openshift.io/cluster` object's registrySources fields
```

Comment:
(BTW callers of the containers/image library can point it at other locations, so the code does not necessarily need root inside the container to be able to overwrite files in `/etc`.)

Reply:
We will not leverage a privileged/rooted pod for changing this folder and will address the actual location of the files at implementation time.